File: C++/686. Repeated String Match.cpp
class Solution {
public:
    int repeatedStringMatch(string A, string B) {
        int cnt = B.length()/A.length()+3;
        string a;
        for(int i = 1;i<=cnt;i++)
        {
            a += A;
            if(a.find(B) != string::npos)
                return i;
        }
        return -1;
    }
};

File: Python/686. Repeated String Match.py
class Solution:
    def repeatedStringMatch(self, A, B):
        """
        :type A: str
        :type B: str
        :rtype: int
        """
        for i in range(len(B) // len(A) + 3):
            if B in A * i:
                return i
        return -1